### PR TITLE
Fix IdIndex map

### PR DESCRIPTION
**Pull request by braktar**

### CODE REVIEW

File: tsptw_data_dt.h [hunk 1]
```diff
@@ -852,7 +852,8 @@ void TSPTWDataDT::LoadInstance(const std::string& filename) {
 
         service_times_.push_back(service.duration());
         alternative_size_map_[service.problem_index()] += 1;
-        ids_map_[(std::string)service.id()] = node_index;
+        if (ids_map_.find((std::string)service.id()) == ids_map_.end())
+          ids_map_[(std::string)service.id()] = node_index;
```
Comment (senhalil) [inline tsptw_data_dt.h:856]: 1. we do this `ids_map_[(std::string)service.id()] = node_index;` exact operation in other places, do we need to fix them as well?
2. what's the meaning of service.id not existing in the map? "not having any alternative" ot that this is the first altearnative or something else?
3. why don't we need to register the node_index if the service_id is in the map?

**Review by senhalil — COMMENTED** [review]
good find!

File: tsptw_data_dt.h (hunk 1 above)
Comment (braktar) [reply tsptw_data_dt.h:856]: 1. The only other place is for cases without multiple soft timewindows and the ortools service id is defined on optim-api as `#{service.id}_activity#{activity_index}`
2. If the service.id does not exist in the map it means we just have met it first occurrence.
3. We only need to register the node_index of the first occurence we store elsewhere the alternative size. The multiple alternative nodes of the same ortools service are consecutive.

Comment (senhalil) [reply tsptw_data_dt.h:856]: Thanks for the explanation, it is more clear now. :rocket:

**Pull request merged**